### PR TITLE
migrate to go module

### PR DESCRIPTION
**Pull request by honnix**
Please treat this as a starting point of discussion than actual PR.

Since everything now is on go-1.3, and go module seems to be mature enough, what do you think of the migration?

### CODE REVIEW

**Comment by honnix** [comment]
@wild-endeavor PTAL

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	github.com/lyft/flyteidl v0.17.0
+	github.com/lyft/flytestdlib v0.3.0
+	github.com/mitchellh/mapstructure v1.1.2
+	github.com/spf13/cobra v0.0.5
+	github.com/spf13/pflag v1.0.5
+	github.com/stretchr/testify v1.4.0
+	google.golang.org/grpc v1.26.0
+)
+
+// Pin the version of client-go to something that's compatible with katrogan's fork of api and apimachinery
```
Comment (chanadian) [inline go.mod]: @wild-endeavor Do we have to do this even though datacatalog doesn't need this fork?

**Comment by wild-endeavor** [comment]
@katrogan could you post some background info here (or link to some) about why your fork of k8s is necessary and what it fixes?

**Comment by katrogan** [comment]
Sigh, yes. I'm trying to upstream changes around quantity proto marshaling which are currently broken on kubernetes master. without these we can't use sidecar plugins. until those changes land we have to use the lyft fork of k8s

File: go.mod (hunk 1 above)
Comment (katrogan) [reply go.mod]: we can use the lyft fork

**Pull request merged**